client/network_create.go

```diff
@@ -13,6 +13,9 @@ func (cli *Client) NetworkCreate(ctx context.Context, name string, options types
 		NetworkCreate: options,
 		Name:          name,
 	}
+	// CheckDuplicate is always set to true to make sure clients talking to old daemons will still get an error
+	// about duplicate networks.
+	networkCreateRequest.CheckDuplicate = true //nolint:staticcheck // ignore SA1019
```

Comment:
Ah, sorry, perhaps I should've mentioned more clearly; should we make this based on API version?

- If API-version < XX, then set it to `true`
- For higher API versions, the daemon already has it as default, so we no longer need to send it

Having it gated by version also helps us find back these kind of things in future (i.e., when we deprecate API < vXX)


I would also suggest including the API version in the `// ignore ..` part of the comment (also to help find back "since when").